app/views/unattended/provisioning_templates/finish/windows_default_finish.erb

```diff
@@ -31,102 +33,108 @@ description: |
+<% if host_param('machinePassword').present? -%>
+echo Performing unsecure domain join
+powershell.exe -OutputFormat text -command Add-Computer -Domain '<%= host_param('computerDomain') -%>' -Options UnsecuredJoin,PasswordPass -Credential (New-Object -TypeName System.Management.Automation.PSCredential $null, (ConvertTo-SecureString -String '<%= host_param('machinePassword') -%>' -AsPlainText -Force))
+<% end -%>
+<% end -%>
+<% end -%>
+
+<% if host_param('localAdminAccountDisabled') -%>
+echo Disabling %tempAdminUser%
+net user %tempAdminUser% %tempAdminUser% /active:no
+<% end -%>
+
+<% if host_param('http-proxy').present? -%>
+cmd /C "netsh winhttp set proxy <%= host_param('http-proxy') %>:<%= host_param('http-proxy-port') %>"
+<% end -%>
+
+<% unless host_param('computerDomain') -%>
+powershell /c "Get-NetConnectionProfile -InterfaceAlias \"Ethernet0\" | Set-NetConnectionProfile -NetworkCategory <%= network_location %>"
+<% end -%>
+
+<% if host_param('ansible_user').present? -%>
+<% if host_param_true?('create_ansible_user') -%>
+powershell /c "set-localuser -name <%= host_param('ansible_user') %> -passwordneverexpires 1"
+<% end -%>
+powershell /c "Enable-PSRemoting"
+cmd /c "netsh advfirewall firewall add rule name="WinRM-HTTP" dir=in localport=5985 protocol=TCP action=allow"
+cmd /c winrm set winrm/config/service @{AllowUnencrypted="true"}
+cmd /c winrm set winrm/config/client/auth @{Basic="true"}
+cmd /c winrm set winrm/config/service/auth @{Basic="true"}
+<% end -%>
+
+<% if host_param('ping') -%>
+cmd /c "netsh advfirewall firewall add rule name=\"Enable IPv4 ICMP\" dir=in protocol=icmpv4 action=allow"
+<% end -%>
+
+<% if host_param('remote_desktop') -%>
+cmd /c "netsh advfirewall firewall set rule group=\"remote desktop\" new enable=Yes"
+cmd /c "netsh advfirewall firewall set rule group=\"remotedesktop\" new enable=Yes"
+<% end -%>
+
+<% if puppet_enabled -%>
+echo Downloading Puppet installer
+wget "<%= host_param('win_puppet_source') %>" -O C:\puppet-agent-x64-latest.msi
+echo Installing Puppet
+start /w "" msiexec /qn /i C:\puppet-agent-x64-latest.msi PUPPET_AGENT_STARTUP_MODE=Manual PUPPET_SERVER=<%= host_puppet_server -%> PUPPET_CA_SERVER=<%= host_puppet_ca_server -%> PUPPET_AGENT_ACCOUNT_DOMAIN=<%= @host.domain -%> PUPPET_AGENT_ACCOUNT_USER=administrator PUPPET_AGENT_ACCOUNT_PASSWORD="<%= host_param('domainAdminAccountPasswd') -%>"
+echo Setting Puppet to auto start
+sc config puppet start= auto
+sc query puppet
+<% end -%>
+
+<% if !host_param_true?('foremanDebug') -%>
+echo Rebooting in 60 sec
+shutdown /r /t 60
+
+echo Removing wimaging files
+rd /s /q c:\wimaging
+sdelete.exe -accepteula -p 2 c:\Windows\Panther\unattend.xml
+sdelete.exe -accepteula -p 2 C:\Windows\Setup\Scripts\SetupComplete.cmd
+
+echo Removing leftover directories
+rd /s /q c:\MININT
+rd /s /q c:\drivers
+rd /s /q c:\updates
+
+<% if puppet_enabled -%>
+echo Removing Puppet installer
+sdelete.exe -accepteula -p 2 C:\puppet-agent-x64-latest.msi
```

Comment:
And if you change the filename above, then here too
```suggestion
sdelete.exe -accepteula -p 2 C:\puppet-agent-latest.msi
```

Reply:
Note: If I change the name here, I also need to change it in the windows user data template.